File: image-service/src/main/java/com/oognuyh/imageservice/util/annotation/CurrentUserId.java
package com.oognuyh.imageservice.util.annotation;

import java.lang.annotation.Documented;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

import org.springframework.security.core.annotation.AuthenticationPrincipal;

@Target({ ElementType.PARAMETER, ElementType.TYPE })
@Retention(RetentionPolicy.RUNTIME)
@Documented
@AuthenticationPrincipal(expression = "#this == 'anonymousUser' ? null : subject")
public @interface CurrentUserId {
    
}

File: notification-service/src/main/kotlin/com/oognuyh/notificationservice/util/annotation/CurrentUserId.kt
package com.oognuyh.notificationservice.util.annotation

import org.springframework.security.core.annotation.AuthenticationPrincipal

@Target(AnnotationTarget.TYPE, AnnotationTarget.VALUE_PARAMETER)
@Retention(AnnotationRetention.RUNTIME)
@MustBeDocumented
@AuthenticationPrincipal(expression = "#this == 'anonymousUser' ? null : subject")
annotation class CurrentUserId
